vertx-web-openapi/src/test/java/io/vertx/ext/web/openapi/RouterBuilderIntegrationTest.java

```diff
@@ -739,6 +741,52 @@ public void testPathParameter(Vertx vertx, VertxTestContext testContext) {
     });
   }
 
+  @Test
+  public void testPathSingleParameterTemplateMatching(Vertx vertx, VertxTestContext testContext) {
+    final Checkpoint checkpoint = testContext.checkpoint(2);
+    final String expectedMessageById = "byId";
+    final String expectedMessageMine = "mine";
+
+    loadBuilderAndStartServer(vertx, VALIDATION_SPEC, testContext, routerBuilder -> {
+      defineOperation("showPetById", expectedMessageById, routerBuilder);
+      defineOperation("showMyPets", expectedMessageMine, routerBuilder);
+    }).onComplete(h -> {
+      testGetRequest200WithMessage(client, "/pets/3", expectedMessageById).send(testContext, checkpoint);
+      testGetRequest200WithMessage(client, "/pets/mine", expectedMessageMine).send(testContext, checkpoint);
+    });
+  }
+
+  @Test
+  public void testPathMultipleParametersTemplateMatching(Vertx vertx, VertxTestContext testContext) {
+    final Checkpoint checkpoint = testContext.checkpoint(3);
+    final String expectedMessagePetByIdInStoreById = "showPetByIdInStoreById";
+    final String expectedMessagePetByIdInStorex = "showPetByIdInStorex";
+    final String expectedMessageDogsInStoreById = "showDogsInStoreById";
+
+    loadBuilderAndStartServer(vertx, VALIDATION_SPEC, testContext, routerBuilder -> {
+      defineOperation("showPetByIdInStoreById", expectedMessagePetByIdInStoreById, routerBuilder);
+      defineOperation("showPetByIdInStorex", expectedMessagePetByIdInStorex, routerBuilder);
+      defineOperation("showDogsInStoreById", expectedMessageDogsInStoreById, routerBuilder);
+    }).onComplete(h -> {
+      testGetRequest200WithMessage(client, "/pets/1/1", expectedMessagePetByIdInStoreById)
+        .send(testContext, checkpoint);
+      testGetRequest200WithMessage(client, "/pets/storex/1", expectedMessagePetByIdInStorex)
+        .send(testContext, checkpoint);
+      testGetRequest200WithMessage(client, "/pets/1/dogs", expectedMessageDogsInStoreById)
+        .send(testContext, checkpoint);
+    });
+  }
+
+  private static void defineOperation(String operationId, String expectedMessage, RouterBuilder routerBuilder) {
+    routerBuilder
+      .operation(operationId)
+      .handler(routingContext -> routingContext.response().setStatusMessage(expectedMessage).end());
+  }
+
+  private static TestRequest testGetRequest200WithMessage(WebClient client, String path, String expectedMessage) {
```

Comment:
Can you avoid creating this method? Just writing `testRequest...` helps the readability